### PR TITLE
[sailfish-browser] Change page background handling to follow page container.  Fixes JB#29662

### PR DESCRIPTION
**Pull request by rainemak**
Review @rojkov @mikkoharju 

Sharing via email and selecting contact ended up being tough nut to crack but now that works as well.

Also sharing via email and picking attachments is working and as well as "normal" facebook like sharing. This doesn't contain external dependencies.

Same approach will be automatically used for web prompts, password manager dialogs, etc.

### CODE REVIEW

File: src/browser.qml [hunk 1]
```diff
@@ -57,4 +93,48 @@ ApplicationWindow {
...
+
+                    var index = 0
+                    for (; index < backgrounds.length; ++index) {
+                        if (backgrounds[index].background === bg) {
+                            break
+                        }
+                    }
+
+                    if (index === 0 && backgrounds.length > 0) {
+                        backgrounds = backgrounds.slice(index + 1)
```
Comment (rojkov) [inline src/browser.qml]: could be just
backgrounds = backgrounds.slice(1)

File: src/browser.qml [hunk 2]
```diff
@@ -57,4 +93,48 @@ ApplicationWindow {
...
+                    } else if (index > 0 && index < backgrounds.length - 1) {
+                        var firstPart = backgrounds.slice(0, index)
+                        var secondPart = backgrounds.slice(index + 1)
+                        backgrounds = firstPart.concat(secondPart)
```
Comment (rojkov) [inline src/browser.qml]: splice?

File: src/browser.qml (hunk 1 above)
Comment (rainemak) [inline src/browser.qml]: True :)

File: src/browser.qml (hunk 2 above)
Comment (rainemak) [inline src/browser.qml]: Good point... Following should be enough.

`background.splice(index, 1)`

Will verify that original array updates correctly.

**Comment by rainemak** [comment]
Rebased and updated.

File: src/browser.qml [hunk 3]
```diff
@@ -48,4 +86,45 @@ ApplicationWindow {
...
+            onPageChanged: {
+                if (!page) {
+                    var index = 0
+                    for (; index < backgrounds.length; ++index) {
+                        if (backgrounds[index].background === bg) {
+                            break
+                        }
+                    }
+
+                    if (index === 0 && backgrounds.length > 0) {
```
Comment (rojkov) [inline src/browser.qml]: I guess that given `index` is initialized to -1 these three conditions could be collapsed to only one

```
if (index >=  0 && backgrounds.length > 0) {
    background.splice(index, 1)
} else {
    backgrounds = []
}
```

Comment (rainemak) [inline src/browser.qml]: That would work. Looks that when I wrote this a late night I wasn't thinking too much :)

Index is initialized to 0 but length > 0 guards first block. So, the else block takes care of the rest. Thanks.

**Comment by rainemak** [comment]
Added one commit on top that fixes content resizing on sub pages. @mikkoharju could give a look. What do you think about having a bit bigger background. As the background is not blended, I didn't see anything extra when peeking (popping) a page from deeper hierarchy.

E.g. Share => Facebook and pop

**Comment by rainemak** [comment]
@antseppa could you check as well.

File: src/browser.qml [hunk 4]
```diff
@@ -48,4 +86,41 @@ ApplicationWindow {
...
+        id: backgroundComponent
+        Browser.Background {
+            id: bg
+            property Item pageContainer
+            property Page page
+
+            // Page and attached background is destroyed when the page gets destroyed.
+            onPageChanged: {
+                if (!page) {
+                    var index = 0
```
Comment (rojkov) [inline src/browser.qml:100]: `index` should be better initialized to `-1`

Comment (rojkov) [inline src/browser.qml:100]: or perhaps that would be too defensive style... LGTM

**Comment by antseppa** [comment]
LGTM, works for me.

**Pull request merged**